app/models/communication.rb

```diff
@@ -60,6 +61,20 @@ def self.create_with_twilio_details!(appointment:, twilio_sid:, twilio_msg_statu
     end
   end
 
+  def self.create_with_imo_details!(appointment:, notification:, result:)
+    transaction do
+      # create detailable here once detailable model exists
+      communication = Communication.create!(communication_type: "imo",
+                                            detailable: nil,
+                                            appointment: appointment,
+                                            notification: notification,
+                                            device_created_at: DateTime.current,
+                                            device_updated_at: DateTime.current)
+      logger.info class: self.class.name, msg: "create_with_imo_details", communication: communication.id,
```

Comment:
Consider renaming the key `communication_id` for consistency.

Reply:
Good catch. This was copy-pasted from the twilio method above. I'll correct it there too.